agent/consul/leader_connect_ca.go

```diff
@@ -534,6 +537,7 @@ func (c *CAManager) initializeSecondaryCA(provider ca.Provider, config *structs.
 		if err != nil {
 			return err
 		}
+		storedRoot = ca.AddSingleNewline(storedRoot)
```

Comment:
same here, despite the name, it doesn't look like we need a trailing newline here